R/helpers.R

```diff
@@ -13,6 +13,43 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+.geocode <- function(name, country, method) {
+  stopifnot("Please provide a location name as a string" = is.character(name))
+
+  # Handle API keys for methods that require them
+  methods_with_keys <- c("google", "bing", "here", "tomtom", "mapbox", "geocodio")
+  if (method %in% methods_with_keys) {
+    api_key_env <- paste0(toupper(method), "_API_KEY")
+    api_key <- Sys.getenv(api_key_env)
+    if (api_key == "") {
+      stop(paste0("API key for ", method, " is required. Please set the '", api_key_env, "' environment variable."))
+    }
+  }
+
+  result <- tryCatch(
+    {
+      geocode_df <- tidygeocoder::geocode(
+        data.frame(address = paste0(name, " ", country)), "address",
+        method = method, quiet = TRUE
+      )
```

Comment:
Uses lower-level function `geo()` with less data wrangling

Reply:
I prefer to use the following: 
```

      geocode_df <- tidygeocoder::geo(
        address = paste0(name, " ", country),
        method = method, quiet = TRUE, progress_bar = FALSE, limit = 1
      )
```
Which allow to pass any location rather than being restricted to city names. Doing so allows to geocode a neighborhood name for example.

Want do you think?

Reply:
> I prefer to use the following:
> 
> ```
> 
>       geocode_df <- tidygeocoder::geo(
>         address = paste0(name, " ", country),
>         method = method, quiet = TRUE, progress_bar = FALSE, limit = 1
>       )
> ```
> 
> Which allow to pass any location rather than being restricted to city names. Doing so allows to geocode a neighborhood name for example.
> 
> Want do you think?

Good point!